src/migrations.ts

```diff
@@ -203,11 +206,22 @@ export class Migration {
     const files = await promisify(fs.readdir)(dirPath);
     let ver = 1;
 
-    const loadedScripts = await this.getLoadedScripts('asc');
-    const notLoadedScripts = files.filter((file) => !loadedScripts.find((loadedScripts) => loadedScripts.fileName === file));
+    const loaded = await this.getLoadedScripts('asc');
+    const notLoaded = files.filter((file) => !loaded.find((loadedScripts) => loadedScripts.fileName === file));
     
-    const sortedNotLoadedScripts = notLoadedScripts.sort(this.sortFiles);
-    const fileArr = [...loadedScripts.map((script) => script.fileName), ...sortedNotLoadedScripts];
+    const sortedNotLoadedScripts = notLoaded.sort(this.sortFiles);
+
+    if(loaded.length && sortedNotLoadedScripts.length) {
+      const lastLoadedScript = loaded[loaded.length - 1];
+      // Get timestamp from filename of both scripts
+      const lastLoadedScriptTimestamp = lastLoadedScript.fileName.match(/^(\d*)-/)[1];
+      const firstNotLoadedScriptTimestamp = sortedNotLoadedScripts[0].match(/^(\d*)-/)[1];
+      if(process.env.MODE === 'prod' && firstNotLoadedScriptTimestamp < lastLoadedScriptTimestamp) {
```

Comment:
Please change env variable name to be more descriptive, so it is less possible to interfere with other packages. For example:
```
MIGRATIONS_STRICT_ORDER = true #/false
# if not set, use "true" when process.env.NODE_ENV == "production" and false otherwise.
```